.pre-commit-hooks.yaml

```diff
@@ -13,6 +13,12 @@
   entry: ruby.py
   language: script
   files: \.(rb|rake)$
+- id: restrict_methods_on_reservation_table
+  name: Only allow limited methods on the Reservation model in parsing
+  entry: ^Reservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*
```

Comment:
I'm afraid this `^` doesn't work with the -z option:
```shell
vagrant@vagrant:~ (@) 14:35:25 $ grep -bzo -P '^Reservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*' test.rb
grep: unescaped ^ or $ not supported with -Pz
```

I think what you would really want to do is use a negative lookbehind to not match `TemporaryReservation`:
For example, given a `test.rb` containing
```ruby
Reservation.something
  Reservation.something
a = Reservation.something
TemporaryReservation.something
```

### Examples without `-zo` for friendlier output here:
#### Old way:
```
vagrant@vagrant:~ (@) 14:38:15 $ grep -b -P 'Reservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*' test.rb
0:Reservation.something
22:  Reservation.something
46:a = Reservation.something
72:TemporaryReservation.something
```

#### With negative lookbehind:
```
vagrant@vagrant:~ (@) 14:39:04 $ grep -b -P '(?<!Temporary)Reservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*' test.rb
0:Reservation.something
22:  Reservation.something
46:a = Reservation.something
```

So I think the regex you want to keep is actually `(?<!Temporary)Reservation\s*\.(?!(mass|calendar|hotel_pms_res|set_canceled|update_from)).*` (the `.*` at the end is again here for output cosmetic purposes)